CoreDataAccessView/test/unit/src/au/gov/asd/tac/constellation/views/dataaccess/plugins/clean/MergeNodesByPrefixNGTest.java

```diff
@@ -0,0 +1,242 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package au.gov.asd.tac.constellation.views.dataaccess.plugins.clean;
+
+import au.gov.asd.tac.constellation.graph.StoreGraph;
+import au.gov.asd.tac.constellation.graph.schema.Schema;
+import au.gov.asd.tac.constellation.graph.schema.SchemaFactoryUtilities;
+import au.gov.asd.tac.constellation.graph.schema.analytic.AnalyticSchemaFactory;
+import au.gov.asd.tac.constellation.graph.schema.visual.concept.VisualConcept;
+import au.gov.asd.tac.constellation.plugins.parameters.PluginParameter;
+import au.gov.asd.tac.constellation.plugins.parameters.PluginParameters;
+import au.gov.asd.tac.constellation.plugins.parameters.types.BooleanParameterType;
+import au.gov.asd.tac.constellation.plugins.parameters.types.IntegerParameterType;
+import static au.gov.asd.tac.constellation.views.dataaccess.plugins.clean.MergeNodesPlugin.LEAD_PARAMETER_ID;
+import static au.gov.asd.tac.constellation.views.dataaccess.plugins.clean.MergeNodesPlugin.MERGER_PARAMETER_ID;
+import static au.gov.asd.tac.constellation.views.dataaccess.plugins.clean.MergeNodesPlugin.MERGE_TYPE_PARAMETER_ID;
+import static au.gov.asd.tac.constellation.views.dataaccess.plugins.clean.MergeNodesPlugin.SELECTED_PARAMETER_ID;
+import static au.gov.asd.tac.constellation.views.dataaccess.plugins.clean.MergeNodesPlugin.THRESHOLD_PARAMETER_ID;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertTrue;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+/**
+ * Unit tests for MergeNodesByPrefix.
+ *
+ * @author sol695510
+ */
+public class MergeNodesByPrefixNGTest {
+
+    private StoreGraph graph;
+    private int vxId1, vxId2, vxId3, vxId4, vxId5;
+    private int vertexIdentifierAttribute, vertexSelectedAttribute;
+
+    private final Comparator<String> LONGEST_VERTEX_CHOOSER = (String o1, String o2) -> {
+        if (o1.length() > o2.length()) {
+            return -1;
+        } else if (o1.length() < o2.length()) {
+            return 1;
+        } else {
+            return o1.compareTo(o2);
+        }
+    };
+
+    private final Comparator<String> SHORTEST_VERTEX_CHOOSER = (String o1, String o2) -> {
+        if (o1.length() > o2.length()) {
+            return 1;
+        } else if (o1.length() < o2.length()) {
+            return -1;
+        } else {
+            return o1.compareTo(o2);
+        }
+    };
```

Comment:
@sol695510 can you please use the instance in `MergeNodesPlugin` and make that protected so that you can access it from this unit test?